archive_viewer/table_models/curve_model.py

```diff
@@ -145,6 +174,90 @@ def set_model_curves(self, curves: List[Dict]) -> None:
         self.endResetModel()
         logger.debug("Finished setting curves model")
 
+    def replaceToArchivePlot(self, curve: BasePlotCurveItem, index: QModelIndex, address: str, color: Optional[QColor] = None):
+        self.append(address=address, name=address, color=color)
+        self._plot._curves[index.row()] = self._plot._curves[-1]
+        self.beginRemoveRows(QModelIndex(), self.rowCount() - 1, self.rowCount() - 1)
+        self.plot._curves = self.plot._curves[:-1]
+        self.plot.removeItem(curve)
+        self.endRemoveRows()
+
+    def formulaToPVDict(self, rowName: str, formula: str) -> dict:
+        pvs = re.findall("{(.+?)}", formula)
+        pvdict = dict()
+        for pv in pvs:
+            # Check if all of the requested rows actually exist
+            if pv not in self._row_names:
+                raise ValueError(f"{pv} is an invalid variable name")
+            elif pv == rowName:
+                raise ValueError(f"{pv} is recursive")
+            elif self._row_names.index(pv) > self._row_names.index(rowName):
+                raise ValueError(f"Error, all referenced curves must come before the Formula")
+            else:
+                # if it's good, add it to the dictionary of curves. rindex = row index (int) as opposed to index, which is a QModelIndex
+                rindex = self._row_names.index(pv)
+                pvdict[pv] = self._plot._curves[rindex]
+        return pvdict
+
+    def replaceToFormula(self, index: QModelIndex, formula: str, color: Optional[QColor] = None) -> bool:
+        """Replaces existing ArchivePlotCurveItem with a new FormulaCurveItem
+
+        Parameters
+        ----------
+        formula : str
+            The Formula we want to graph
+        name : str, optional
+            The display name for the curve.
+        color : Optional[QColor], optional
+            The curve's color on the plot.
+        """
+        # Find row headers using regex
+
+        rowName = self._row_names[index.row()]
+        pvdict = self.formulaToPVDict(rowName, formula)
+        if pvdict == None:
+            return False
+        curve = self._plot._curves[index.row()]
+        if not color:
+            color = ColorButton.index_color(index.row())
+        #          KLYS:LI22:31:KVAC
+        # Handle Archives and formulas differently
+        if index.row() == self.rowCount() - 1:
+            self.append()
+        y_axis = self._axis_model.get_axis(-1)
+        FormulaCurve = FormulaCurveItem(formula=formula, name=formula, pvs=pvdict, color=color, yAxisName=y_axis.name)
+        self._plot._curves[index.row()] = FormulaCurve
+        FormulaCurve.formula_invalid_signal.connect(partial(self.invalidFormula, header = rowName))
+        # Need to check if Formula is referencing a dead row
+        FormulaCurve.redrawCurve()
```

Comment:
Not sure if this error belongs here or in https://github.com/slaclab/pydm/pull/1095
When setting the formula to blank text, whether by typing "f://" as the channel or open FormulaDialog and click "OK", a syntax error gets thrown here by the `redrawCurve()`.

Here is the full traceback:
```
Traceback (most recent call last):
  File "/afs/slac.stanford.edu/u/cd/zdomke/workspace/PyDMArchiveViewer/Archive-Viewer/archive_viewer/mixins/traces_table.py", line 261, in accept_formula
    passed = self.curveModel.replaceToFormula(index = self.curveModel.index(self.parent().selected_index.row(), 0), formula = formula)
  File "/afs/slac.stanford.edu/u/cd/zdomke/workspace/PyDMArchiveViewer/Archive-Viewer/archive_viewer/table_models/curve_model.py", line 232, in replaceToFormula
    FormulaCurve.redrawCurve()
  File "/afs/slac.stanford.edu/u/cd/zdomke/workspace/pydm/pydm/widgets/archiver_time_plot.py", line 553, in redrawCurve
    y = [eval(self._trueFormula), eval(self._trueFormula)]
  File "<string>", line 0

SyntaxError: invalid syntax
```

Reply:
idk why someone would want to do this, but it is a strange edge case

Reply:
Fixed